src/PaletteCell.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2023 Inclusive Design Research Centre, OCAD University
+ * All rights reserved.
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/inclusive-design/adaptive-palette/blob/main/LICENSE
+ */
+
+import { html } from "htm/preact";
+import "./PaletteCell.scss";
+
+export function PaletteCell (props) {
+
+  // Basic styles are the `paletteCell` class defined in PaletteCell.css.
+  // Concatenate any additional classes provided by `props`.
+  let classes = "paletteCell";
+  if (props.class) {
```

Comment:
There isn't yet `class` and `styles` property. Would it be too early to write code to handle them?

Reply:
I'd say it was too early to remove it :-)

I am using an extra `class` for when the cell is disabled, and this use is shown in the tests, `PaletteCell.test.ts`, like so:

```
<${PaletteCell} id="${id}" options=${cellOptions} class="disabled" />
```

That is handled within `PaletteCell` by:
- setting the `button`'s disabled attribute (`<button disabled ...>`)
- adding the `disabled` class to the `button` so the disabled styles are used ( (`<button disabled class="paletteCell disabled" ...>`

However, having said that, I don't know how disabled is actually handled on the palette.  We know that as bmw codes are entered, certain keys are disabled since they are irrelevant for the current code sequence.  How is that handled by Preact?  This is a case of tracking state and probably involves hooks but I do not know the details.  I will have a look at the documentation.